### ISSUE TITLE
Create operator cannot be parsed in postgresql

### ISSUE THREAD

**Issue by kumekay** [sqlfluff/sqlfluff#5111]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Parsing error on `CREATE OPERATOR`

### Expected Behaviour

It should be parsed 

### Observed Behaviour

parsing error: Found unparsable section: 'CREATE OPERATOR >=(/n    LEFTARG = semant...

### How to reproduce

```
CREATE OPERATOR >=(
    LEFTARG = semantic_version,
    RIGHTARG = semantic_version,
    PROCEDURE = semantic_version_ge,
    COMMUTATOR = <=
);
```

### Dialect

postgres

### Version

2.3.0

### Configuration

[tool.sqlfluff.core]
dialect = 'postgres'
max_line_length = 100

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by R3gardless** [comment]
@alanmcruickshank
According to the [official documentation](https://www.postgresql.org/docs/current/sql-createoperator.html), there are certain patterns to follow when creating operators. 
However, after creating a new operator, using it results in a parse error.

It seems like modifying other segment rules might resolve this issue. 
But, if I modify those rules, it might confuse someone unfamiliar with the operator pattern, making them think it's parsable.

In this case, is it better to just implement the operator creation, or should I modify other segment rules as well?

```SQL
CREATE OPERATOR @===== (
    LEFTARG = box,
    RIGHTARG = box,
    FUNCTION = area_equal_function,
    COMMUTATOR = @=====,
    NEGATOR = !==,
    RESTRICT = area_restriction_function,
    JOIN = area_join_function,
    HASHES, MERGES
);

-- The following code will cause a parse error
SELECT 1 @===== 2;
```

Without the `CREATE OPERATOR` statement:

```SQL
-- Would this cause any parse error?
SELECT 1 @===== 2;
```

**Comment by R3gardless** [comment]
Hi @alanmcruickshank,

I'm currently working on this issue but facing some difficulties.
 I created new lexer rules following the Postgres GitHub [scan.l](https://github.com/postgres/postgres/blob/master/src/backend/parser/scan.l#L382) and [official documentation](https://www.postgresql.org/docs/current/sql-createoperator.html).

Here's my implementation:

```python
postgres_dialect.insert_lexer_matchers(
    [
        RegexLexer(
            "json_operator",
            r"->>?|#>>?|@[>@?]|<@|\?[|&]?|#-",
            SymbolSegment,
        ),
        RegexLexer(
            "postgis_operator",
            r"\&\&\&|\&<\||<<\||@|\|\&>|\|>>|\~=|<\->|\|=\||<\#>|<<\->>|<<\#>>",
            SymbolSegment,
        ),
        RegexLexer(
            "postgres_operator",
            r"(?!^[+\-*\/<>=~!@#%^&|`?]*(--|/\*))(?!^=>$)(?!^[+\-*\/<>=]*[+-]$)(^[+\-*\/<>=~!@#%^&|`?]{1,63})",
            SymbolSegment,
        ),
    ],
    before="word",
)
```

However, after making these changes, some test SQL files are now failing to parse:

- json_operators.sql
- range_operators.sql
- postgres_postgis_operators.sql

One thing I'm unsure about is why the lexer for operators is defined before `like_operator`(Current implementation). Could you give me some advice on how to resolve this issue?

Thanks in advance!
